File: Two Pointers/Valid Palindrome/validPalindrome.rs
impl Solution {
    pub fn is_palindrome(s: String) -> bool {
        let cleaned = s
            .chars()
            .filter(|c| c.is_ascii_alphanumeric())
            .collect::<String>()
            .to_lowercase();
        let reversed = cleaned.chars().rev().collect::<String>();
        cleaned == reversed
    }
}

File: Two Pointers/Valid Palindrome/validPalindrome.cpp
class Solution {
  public:
    bool isPalindrome(string s) {
        string cleaned = "";
        for (auto i : s) {
            if (isalnum(i))
                cleaned.push_back(tolower(i));
        }

        int l = 0;
        int r = cleaned.length() - 1;

        while (l < r) {
            if (cleaned[l] != cleaned[r])
                return false;
            l++;
            r--;
        }
        return true;
    }
};
